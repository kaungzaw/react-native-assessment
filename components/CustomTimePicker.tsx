import { useState } from "react";
import { StyleSheet, Pressable } from "react-native";
import DateTimePickerModal from "react-native-modal-datetime-picker";
import dayjs from "dayjs";
import { ThemedView } from "./ThemedView";
import { ThemedText } from "./ThemedText";
import useFilterStore from "@/store/filterStore";

export default function CustomTimePicker() {
  const [visible, setVisible] = useState(false);
  const { time, setTime } = useFilterStore();

  const handleConfirm = (value: Date) => {
    setTime(dayjs(value));
    hide();
  };

  const show = () => {
    setVisible(true);
  };

  const hide = () => {
    setVisible(false);
  };

  return (
    <>
      <Pressable
        style={({ pressed }) => [pressed ? { opacity: 0.5 } : {}]}
        onPress={show}
      >
        <ThemedText style={styles.label}>Time</ThemedText>
        <ThemedView style={styles.input}>
          <ThemedText>{time?.format("h:mm A") ?? "Select Time"}</ThemedText>
        </ThemedView>
      </Pressable>
      <DateTimePickerModal
        isVisible={visible}
        date={time?.toDate() ?? dayjs("2024-10-10 08:00").toDate()}
        minimumDate={dayjs("2024-10-10 08:00").toDate()}
        maximumDate={dayjs("2024-10-10 19:30").toDate()}
        minuteInterval={30}
        mode="time"
        locale="en-US"
        onConfirm={handleConfirm}
        onCancel={hide}
      />
    </>
  );
}

const styles = StyleSheet.create({
  label: { color: "grey" },
  input: {
    backgroundColor: "transparent",
    borderBottomColor: "lightgrey",
    borderBottomWidth: 1,
    paddingBottom: 6,
  },
});
