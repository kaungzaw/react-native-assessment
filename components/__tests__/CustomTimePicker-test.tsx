import * as React from "react";
import { render } from "@testing-library/react-native";

import CustomTimePicker from "../CustomTimePicker";

it(`renders correctly`, () => {
  const tree = render(<CustomTimePicker />).toJSON();

  expect(tree).toMatchSnapshot();
});

describe("CustomTimePicker", () => {
  test("CustomTimePicker renders correctly", () => {
    const { getByText } = render(<CustomTimePicker />);

    getByText("Time");
  });
});
