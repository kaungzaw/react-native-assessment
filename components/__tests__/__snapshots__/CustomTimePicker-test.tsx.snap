// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {},
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "color": "grey",
        },
      ]
    }
  >
    Time
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "backgroundColor": "transparent",
          "borderBottomColor": "lightgrey",
          "borderBottomWidth": 1,
          "paddingBottom": 6,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Select Time
    </Text>
  </View>
</View>
`;
